control-plane/connect-inject/endpoints_controller.go

```diff
@@ -302,6 +357,17 @@ func getServiceCheck(client *api.Client, healthCheckID string) (*api.AgentCheck,
 	return checks[healthCheckID], nil
 }
 
+// getServiceNameChecks will return all of the health checks for a service.
+func getServiceNameChecks(client *api.Client, serviceName string) (map[string]*api.AgentCheck, error) {
+	filter := fmt.Sprintf("ServiceName == %q", serviceName)
+	checks, err := client.Agent().ChecksWithFilter(filter)
+	if err != nil {
+		return nil, err
+	}
+	// This will be nil (does not exist) or an actual check.
+	return checks, nil
```

Comment:
Only if you are feeling spicy.

```suggestion
	return client.Agent().ChecksWithFilter(filter)
```